Documentation for tags

eshurakov opened the pull request:
In this PR I'm adding documentation for tags. It's the first version with the information that I was able to find about how tags work.

In addition to this documentation I think it would be nice to have a list of test case scenarios as the core behaviour should be the same among all the platforms.

Please take a look and let me know what you think, thank you!

theck13 reviewed (commented):
The documentation looks good to me.  I thought we decided to use the wiki rather than `.md` files.  Did I misunderstand that decision?

There are steps from Android pull requests we can use for test scenarios.
- [Tag renaming](https://github.com/Automattic/simplenote-android/pull/1112#)
- [Canonical representation](https://github.com/Automattic/simplenote-android/pull/1192)
- [Tag submission/validation](https://github.com/Automattic/simplenote-android/pull/1098)

eshurakov commented:
@theck13 I understood that we want to use `.md` files, so that docs can be downloaded and browsed offline. We can see how it goes, it should be easy to migrate to the wiki if needed.

Thank you for the test cases. I changed android specific terms and added them to a separate file.

belcherj reviewed (approved):
Merge it
